bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java

```diff
@@ -0,0 +1,340 @@
+		if (analysisResults.isEmpty()) {
+			LOGGER.log(Level.WARNING, String.format("No analysis results found for obj id = %d", obj_id));
+			return new Score(Significance.UNKNOWN, Confidence.NONE);
+		}
+
+		// find the highest score
+		Score newScore = analysisResults.stream()
+				.map(result -> result.getScore())
+				.max(Score.getScoreComparator())
+				.get();
+
+		// If the new score is diff from current score
+		if ((currentScore.compareTo(new Score(Significance.UNKNOWN, Confidence.NONE)) == 0)
+				|| (Score.getScoreComparator().compare(newScore, currentScore) != 0)) {
+
+			setFinalScore(obj_id, newScore, transaction);
+
+			// fire an event
+			db.fireTSKEvent(new FinalScoreChangedEvent(obj_id, newScore));
+
+			return newScore;
+		} else {
+			// return te current score
+			return currentScore;
+		}
+	}
+
+	/**
+	 * Updates the score for the specified object, if the given analysis result
+	 * score is higher than the score the object already has.
+	 *
+	 * @param obj_id      Object id.
+	 * @param resultScore Score for newly added analysis result.
+	 *
+	 * @return Final score for the object.
+	 *
+	 * @throws TskCoreException
+	 */
+	Score updateFinalScore(long obj_id, Score resultScore) throws TskCoreException {
+		CaseDbTransaction transaction = db.beginTransaction();
+		try {
+			Score score = updateFinalScore(obj_id, resultScore, transaction);
+
+			transaction.commit();
+			return score;
+		} catch (TskCoreException ex) {
+			try {
+				transaction.rollback();
+			} catch (TskCoreException ex2) {
+				LOGGER.log(Level.SEVERE, "Failed to rollback transaction after exception. "
+						+ "Error invoking updateScore with obj_id: " + obj_id, ex2);
+			}
+			throw ex;
+		}
+	}
+
+	/**
+	 * /**
+	 * Updates the score for the specified object, if the given analysis result
+	 * score is higher than the score the object already has.
+	 *
+	 * @param obj_id      Object id.
+	 * @param resultScore Score for a newly added analysis result.
+	 * @param transaction Transaction to use for the update.
+	 *
+	 * @return Final score for the object.
+	 *
+	 * @throws TskCoreException
+	 */
+	Score updateFinalScore(long obj_id, Score resultScore, CaseDbTransaction transaction) throws TskCoreException {
+
+		// Get the current score 
+		Score currentScore = getFinalScore(obj_id, transaction);
+
+		// If the current score is Unknown or the new score is higher than the current score
+		if ((currentScore.compareTo(new Score(Significance.UNKNOWN, Confidence.NONE)) == 0)
+				|| (Score.getScoreComparator().compare(resultScore, currentScore) > 0)) {
+
+			setFinalScore(obj_id, resultScore, transaction);
+
+			// fire an event
+			db.fireTSKEvent(new FinalScoreChangedEvent(obj_id, resultScore));
+			return resultScore;
+		} else {
+			// return te current score
+			return currentScore;
+		}
+	}
+
+	/**
+	 * Event fired to indicate that the score of an object has changed. 
+	 */
+	final public class FinalScoreChangedEvent {
```

Comment:
Is there much value in including the old score?  The score is updated only if it is different from the previous score.  

We are actually changing this event to include a map<obj_id, newScore> so that it can be used to signal a change in score for multiple objects at the same time.

Reply:
By putting the old score in the events, we enable subscribers to decide whether the score change is an update or to handle different transitions. It also seems kind of standard. What is the cost of dong so? Is this more than just adding a field to the event?